controller.py

```diff
@@ -401,74 +431,139 @@ def add_with_metadata(self, collection_details):
 
         feed = feedparser.parse(request.data)
         entries = feed.get("entries", [])
+        entries_by_urn = { entry.get('id') : entry for entry in entries }
 
-        for entry in entries:
-            urn = entry.get('id')
-            try:
-                identifier, ignore = Identifier.parse_urn(
-                    self._db, urn
-                )
-            except Exception as e:
-                identifier = None
+        identifiers_by_urn, invalid_urns = Identifier.parse_urns(
```

Comment:
I wanted to start using bulk parsing here, since we've changed it in other places in the controller and metadata requests may result in more engagement here.